### PR TITLE
When selecting a disabled period, redirect to the default "yesterday" period

### LINKED ISSUE #7615: Unable to log into Piwik after disabling ranges as period in UI
How to reproduce:
- log into Piwik, set for ex. "last7" as default date, log out,
- set ```enabled_periods_UI = day,week,month,year```
- try logging back into Piwik - you should be served an error message saying that period is invalid.
I guess that in such case, user setting should be ignored and "yesterday" loaded instead or something like that.

Please let me know if this is enough to reproduce issue

### PR DESCRIPTION
**Pull request by mnapoli**
Fixes #7615

I redirected to hardcoded "yesterday" because I don't want to overthink it, but if you think it's a bad idea let me know.

### CODE REVIEW

**Comment by mnapoli** [comment]
FYI build failed because of random failure: http://builds-artifacts.piwik.org/ui-tests.rang-disabled/11472.7/screenshot-diffs/diffviewer.html

**Comment by mgazdzik** [comment]
@mnapoli - maybe it could be a config setting defaulting to yesterday? I'm preety much confident that there will be a case at some point of future when somebody will prefer to set different fallback than 'yesterday'

**Comment by mattab** [comment]
There are config settings: [General] `default_day` and `default_period` so let's re-use those

**Comment by mnapoli** [comment]
PR updated

**Comment by tsteur** [comment]
Does it maybe make sense to fix this in the `getDefaultPeriod` and `getDefaultDate` method itself? https://github.com/piwik/piwik/blob/2.13.0-b1/plugins/UsersManager/UserPreferences.php#L75-L120 

I think the error message / exception is ok if someone manipulates that URL directly etc. Also if we use eg `getDefaultPeriod` somewhere else we would otherwise have this problem again. It would maybe also make sure to show correct value in the "User Settings" UI (eg yesterday instead of "last 7" or instead of no preselected value, haven't actually tested it)

**Comment by mnapoli** [comment]
Yes thanks it definitely makes sense, I will update the PR for that solution

**Comment by mnapoli** [comment]
I have updated the PR with an alternate solution that prevents UserPreferences to return a disallowed period or date. I have also restored the exception.

File: plugins/UsersManager/UserPreferences.php [hunk 1]
```diff
@@ -74,8 +85,45 @@ public function getDefaultReport()
...
+    {
+        list($defaultDate, $defaultPeriod) = $this->getDefaultDateAndPeriod($defaultDate);
+
+        return $defaultPeriod;
+    }
+
+    private function getDefaultDateAndPeriod($defaultDate = null)
+    {
+        if (! $defaultDate) {
+            $defaultDate = $this->getDefaultPeriodWithoutValidation($defaultDate);
```
Comment (mattab) [inline plugins/UsersManager/UserPreferences.php]: `getDefaultPeriodWithoutValidation` should be `getDefaultDateWithoutValidation`

**Comment by mattab** [comment]
The refactoring/solution looks good to me (there is one feedback to fix and it  can merged then)

File: plugins/UsersManager/UserPreferences.php (hunk 1 above)
Comment (mnapoli) [inline plugins/UsersManager/UserPreferences.php]: reproduced with tests and fixed

**Comment by mnapoli** [comment]
Fixed, good to merge

**Pull request merged**